src/App.js

```diff
@@ -1,25 +1,29 @@
-import logo from './logo.svg';
+import React, { useState } from 'react';
 import './App.css';
+import Login from './Login'; 
+import mockUsers from '../src/mockusers'; 
 
-function App() {
+const App = () => { 
+  const [loggedInUser, setLoggedInUser] = useState(null);
+
+  const handleLogin = (username, password) => {
+    const user = mockUsers.find(
+      (user) => user.username === username && user.password === password
+    );
+    if (user) {
+      setLoggedInUser(user.username);
+    } else {
+      alert('Invalid username or password');
```

Comment:
This should be a message on the DOM and not an alert. Think of how other sites deal with incorrect passwords.

Reply:
I added a state variable in the Login component to track the login status. Instead of using an alert, I rendered the error message on the Login form itself.

Reply:
A (<p>) tag  with the class 'error-message' will be rendered below the login button when there is an error during login. If the loginError state is not empty, it will show the error message on the form, otherwise, it will not display anything. I feel this was a to render the error message in the DOM without using the 'alert function'.

now just to style the error message on the DOM for Readability.